### ISSUE TITLE
Add a `check_turn_damage` function to mech instrument panel

### ISSUE THREAD

**Issue by autodidaddict** [wascc/wasmdome#10]
Mechs should be able to check and see if, during the last round, they were attacked/damaged by another mech or entity. This kind of information is helpful to inform navigation and targeting algorithms. For example:
* If the mech was damaged by another mech last turn, the mech can adjust its strategy accordingly
* If the mech took damage from collision during the last turn, the mech can make a decision about its navigation strategy

To implement this, I suggest the following:
* Modify `domaincommon` event sourcing rules such that whenever a mech takes damage from anything, metadata about the thing that damaged it be stored in a slot. This slot should only last 1 turn, so this query function should only ever return a value if the damage took place on the previous turn.
* Modify the mech SDK to add a `check_turn_damage` that returns an `Option` containing the source of, and amount of, damage taken during the last turn. (this can be done by reading the mech state that gets delivered to each mech during its "take turn" cycle).

**Comment by autodidaddict** [comment]
Maybe change this to `last_damage_source()` , which returns an option. This could remain as it is until the source of damage changes, and then it's up to the mech to deal with that information as they choose. I'm just having a gut feeling that the "value that disappears after 1 turn" will be both a source of difficult coding for us and potential friction for mech developers/players.

Thoughts @brooksmtownsend ?

**Comment by brooksmtownsend** [comment]
> 
> 
> Maybe change this to `last_damage_source()` , which returns an option. This could remain as it is until the source of damage changes, and then it's up to the mech to deal with that information as they choose. I'm just having a gut feeling that the "value that disappears after 1 turn" will be both a source of difficult coding for us and potential friction for mech developers/players.
> 
> Thoughts @brooksmtownsend ?

Agreed, 1 turn at a time might get complicated. I'm a fan of the `last_damage_source()` option. If we're including the source and amount of damage in the result, we could also include the turn that the damage occurred on for more information. I'm not exactly sure how that would be useful but while we're adding the feature it might assist some mech designs

**Comment by autodidaddict** [comment]
Yeah if we include the turn when the damage occurred we should also expose the current turn through the instruments trait.
